### PR TITLE
Add Yarn workspace with TypeScript project setup

### PR DESCRIPTION
**Pull request by garyttierney**

### CODE REVIEW

**Comment by kenoir** [comment]
This just for lambdas right? Components are going over in https://github.com/wellcomecollection/wellcomecollection.org I thought.

**Comment by garyttierney** [comment]
> This just for lambdas right? Components are going over in wellcomecollection/wellcomecollection.org I thought.

Yep, the lambdas and auth0 action scripts (and common code for server-side interactions).

File: .buildkite/pipeline.deploy.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+steps:
+
+  # Build the build image.
```
Comment (kenoir) [inline .buildkite/pipeline.deploy.yml:3]: Like splitting the deploy pipeline out 👍

File: .buildkite/scripts/auth0-script-packaging.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+########################################################
+# Script name : init-environment.sh
```
Comment (kenoir) [inline .buildkite/scripts/auth0-script-packaging.sh:3]: nit: `auth0-script-packaging.sh`

File: .buildkite/scripts/linting.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+########################################################
+# Script name : init-environment.sh
```
Comment (kenoir) [inline .buildkite/scripts/linting.sh:3]: nit: `linting.sh`

File: .buildkite/scripts/aws-lambda-packaging.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+
+function __package_lambda() {
+  local package_path="$1"
```
Comment (kenoir) [inline .buildkite/scripts/aws-lambda-packaging.sh:5]: I'd be tempted to make the packaging and deployment of lambdas a task that you run via yarn like: https://github.com/wellcomecollection/wellcomecollection.org/blob/master/.buildkite/pipeline.yml#L81

**Review by danielgrant — APPROVED** [review]

**Pull request merged**